applications/asset_tracker_v2/src/cloud/cloud_codec/lwm2m/lwm2m_codec.c

```diff
@@ -125,18 +115,27 @@ static double pressure_max_range_val = BME680_PRESSURE_MAX_RANGE_VALUE;
 /* Module event handler.  */
 static cloud_codec_evt_handler_t module_evt_handler;
 
-/* Convenience API used to generate path lists with reference to objects that will be sent. */
-static int object_path_list_add(struct cloud_codec_data *output, const char *path)
+/* Convenience API used to generate path lists with reference to objects.
+ * Strings in the path array are expected to be NULL terminated.
+ */
+static int object_path_list_add(struct cloud_codec_data *output, const char * const path[],
```

Comment:
Added a few more run-time checks. In general I like to have APIs return an error code and let the application handle the error and shutdown in a graceful manner instead of using run-time asserts.

Reply:
Thanks for adding the checks!